### DISCUSSION TITLE
How to generate flaws of different inclination angle

### DISCUSSION THREAD

**Discussion by Ali141272** [neperfepx/neper#652] [Q&A] [answered]
Hello Professor Dr. Romain Quey,
Professor is it possible to generate flaws of different inclination angle (such as, α = 00, α = 300, α = 450, α = 600, α = 900) in Neper?
The explanation of the model is as follow:
The surface flaw length is 2a = 10 mm and the width is 1.2 mm and the inclination angle was varied (α = 00, α = 300, α = 450, α = 600, α = 900).
Please see the details in the pictures.

Thank you.
<img width="496" alt="67CA7661-A842-45e5-A0BB-ACD2BBE024F8" src="https://user-images.githubusercontent.com/102291665/222705729-a3a92799-b283-4a5b-affa-58d06c258888.png">
<img width="488" alt="B0C88297-B290-4a10-8110-BB4420AFAAF2" src="https://user-images.githubusercontent.com/102291665/222705774-95734c32-c5b7-4a45-ab54-5f1ac42bbb52.png">

**Comment by rquey** [comment, marked as answer]
Neper itself does not allow this, but I think that some codes may allow to introduce flaws in the polycrystal (Rhino?).

**Reply by Ali141272** [reply]
Professor, Thank you for your kind response.
How to generate different grain sizes model such as Fine grain (FG), Medium grain (MG0 and Coarse grain (CG)? as shown in Fig.1
The description of Fig.1 is as follow:
The size of the models is 22.5 × 45 mm. There is total 500 number of grains in the model as shown in Fig.1.Three different specimens are generated by varying the average edge length in Voronoi tessellation procedure (Fig. 1), which are 1, 1.5 and 2 mm for FG, MG and CG, respectively.

Homogeneity of Grain Size
Is it possible to generate homogeneity of grain size in Neper as you can see in Fig.2?
The sorting coefficient (So) is employed in the same model as shown in Fig.1, to represent the degree of homogeneity of the grain size distribution.

Thank you.

![Picture1](https://user-images.githubusercontent.com/102291665/222910872-1dedb6d8-bab9-43ee-9071-fd0f7033e734.png)
![Picture2](https://user-images.githubusercontent.com/102291665/222910894-06d86f8b-d7d1-424c-bfcf-64d2f114ef5b.png)
